src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -101,15 +105,15 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     Payload payload = GoogleIdHelper.verifyId(request);
     if (payload == null) {
         response.setStatus(400);
         return;
     }
     String userId = payload.getSubject();
-    String userNickname = (String) payload.get(NAME);
-    
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); 
+    String userNickname = (String) payload.get("name");
+        
 
     Long currentChallengeId = 0L;
     // @Erick Change to get length of challenges (replace the 3)
```

Comment:
May i have clarification  on this comment

Reply:
The line `ArrayList<Integer> challengeStatuses = new ArrayList<Integer>(Collections.nCopies(3, 0));` initializes challengeStatuses to an array of three 0s to represent that the user hasn't started any challenges but the length 3 is arbitrary/just based on the fact I wrote 3 mock challenges so you can change the length to how many challenges you want stored.